redash/handlers/query_results.py

```diff
@@ -154,6 +156,20 @@ def options(self, query_id=None, query_result_id=None, filetype='json'):
 
         return make_response("", 200, headers)
 
+    def _fetch_rows(self, query_id):
+        query = models.Query.get_by_id_and_org(query_id, self.current_org)
+        query_result = models.QueryResult.get_by_id_and_org(query.latest_query_data_id, self.current_org)
```

Comment:
🤔 do we need to check execution permission to this query here?

Reply:
The implication of _not_ checking for execution permissions here is that someone could potentially check that specific values are within the result set. Also, this is restricted within the organization. However, it's probably better to double check.